🎉  Source Pinterest: filters for entity statuses

drewrasm opened the pull request:
## What
Previously there was no way to filter off of `entity_statuses` in a Pinterest integration. [Here are the docs for the url params](https://developers.pinterest.com/docs/api/v5/#operation/ads/list).

## How
In the source for Pinterest, the option to allow a status for the entities that can be filtered off of statuses was added. 

<img width="895" alt="Screen Shot 2022-07-26 at 10 12 32 AM" src="https://user-images.githubusercontent.com/44178869/181298984-fae2c46a-6d60-4cfa-9936-4354a985a474.png">

## Recommended reading order
1. `spec.json`
2. `source.py`

## 🚨 User Impact 🚨
The parameters are totally optional, so there is no impact on current functionality.

## Pre-merge Checklist

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

```
Building Airbyte Sub Build: PLATFORM
Type-safe dependency accessors is an incubating feature.

> Task :airbyte-config:specs:generateSeedConnectorSpecs
2022-07-27 16:34:51 INFO i.a.c.EnvConfigs(getEnvOrDefault):967 - Using default value for environment variable SPEC_CACHE_BUCKET: 'io-airbyte-cloud-spec-cache'
2022-07-27 16:34:51 INFO i.a.c.s.SeedConnectorSpecGenerator(run):78 - Updating seeded SOURCE definition specs if necessary...
2022-07-27 16:34:51 INFO i.a.c.s.SeedConnectorSpecGenerator(run):90 - Finished updating /Users/Andrew/projects/airbyte/airbyte-config/init/src/main/resources/seed/source_specs.yaml
2022-07-27 16:34:51 INFO i.a.c.s.SeedConnectorSpecGenerator(run):78 - Updating seeded DESTINATION definition specs if necessary...
2022-07-27 16:34:51 INFO i.a.c.s.SeedConnectorSpecGenerator(run):90 - Finished updating /Users/Andrew/projects/airbyte/airbyte-config/init/src/main/resources/seed/destination_specs.yaml

> Task :airbyte-tests:pmdAcceptanceTests
Execution optimizations have been disabled for task ':airbyte-tests:pmdAcceptanceTests' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/Users/Andrew/projects/airbyte/airbyte-tests/build/resources/acceptanceTests'. Reason: Task ':airbyte-tests:pmdAcceptanceTests' uses this output of task ':airbyte-tests:copyComposeFileForAcceptanceTests' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.4/userguide/validation_problems.html#implicit_dependency for more details about this problem.

> Task :airbyte-tests:pmdAutomaticMigrationAcceptanceTest
Execution optimizations have been disabled for task ':airbyte-tests:pmdAutomaticMigrationAcceptanceTest' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/Users/Andrew/projects/airbyte/airbyte-tests/build/resources/automaticMigrationAcceptanceTest'. Reason: Task ':airbyte-tests:pmdAutomaticMigrationAcceptanceTest' uses this output of task ':airbyte-tests:copyComposeFileForMigrationAcceptanceTests' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.4/userguide/validation_problems.html#implicit_dependency for more details about this problem.

> Task :airbyte-webapp:validateLinks

> airbyte-webapp@0.39.37-alpha validate-links
> ts-node --skip-project ./scripts/validate-links.ts

✓ [tutorialLink] https://www.youtube.com/watch?v=Rcpt5SVsMpk&feature=emb_logo returned HTTP 200
✓ [docsLink] https://docs.airbyte.com returned HTTP 200
✓ [configurationArchiveLink] https://docs.airbyte.com/operator-guides/upgrading-airbyte/ returned HTTP 200
✓ [supportTicketLink] https://airbyte.com/contact-support returned HTTP 200
✓ [helpLink] https://airbyte.com/community returned HTTP 200
✓ [privacyLink] https://airbyte.com/privacy-policy returned HTTP 200
✓ [termsLink] https://airbyte.com/terms returned HTTP 200
✓ [updateLink] https://docs.airbyte.com/operator-guides/upgrading-airbyte returned HTTP 200
✓ [webpageLink] https://airbyte.com returned HTTP 200
✓ [productReleaseStages] https://docs.airbyte.com/project-overview/product-release-stages returned HTTP 200
✓ [statusLink] https://status.airbyte.io/ returned HTTP 200
✓ [normalizationLink] https://docs.airbyte.com/understanding-airbyte/connections#airbyte-basic-normalization returned HTTP 200
✓ [syncModeLink] https://docs.airbyte.com/understanding-airbyte/connections returned HTTP 200
✓ [contactSales] https://airbyte.com/talk-to-sales returned HTTP 200
✓ [namespaceLink] https://docs.airbyte.com/understanding-airbyte/namespaces returned HTTP 200
✓ [gitLink] https://docs.airbyte.com/quickstart/deploy-airbyte returned HTTP 200
✓ [technicalSupport] https://docs.airbyte.com/troubleshooting/on-deploying returned HTTP 200
✓ [demoLink] https://demo.airbyte.io returned HTTP 200
✓ [slackLink] https://slack.airbyte.com returned HTTP 200
✓ [recipesLink] https://airbyte.com/recipes returned HTTP 200

✓ All URLs have been checked successfully.

> Task :airbyte-webapp:test

> airbyte-webapp@0.39.37-alpha pretest
> npm run generate-client


> airbyte-webapp@0.39.37-alpha generate-client
> orval

🍻 Start orval v6.8.1 - A swagger client generator for typescript
🎉 api - Your OpenAPI spec has been converted into ready to use orval!

> airbyte-webapp@0.39.37-alpha test
> craco test "--watchAll=false" "--silent"

PASS src/hooks/services/Feature/FeatureService.test.tsx
PASS src/core/i18n/I18nProvider.test.tsx
PASS src/utils/useTranslateDataType.test.tsx
PASS src/core/jsonSchema/schemaToYup.test.ts
PASS src/hooks/services/Experiment/ExperimentService.test.tsx
PASS src/core/form/uiWidget.test.ts
PASS src/views/Connection/ConnectionForm/calculateInitialCatalog.test.ts
PASS src/core/jsonSchema/schemaToUiWidget.test.ts
PASS src/core/jsonSchema/utils.test.ts
PASS src/hooks/useTrackAction.test.tsx
PASS src/utils/common.test.ts
PASS src/utils/utmStorage.test.ts
PASS src/config/configProviders.test.ts
PASS src/components/StatusIcon/StatusIcon.test.tsx
PASS src/components/ArrayOfObjectsEditor/components/EditorHeader.test.tsx
PASS src/packages/cloud/views/auth/components/GitBlock/GitBlock.test.tsx
PASS src/components/base/Input/Input.test.tsx
PASS src/hooks/services/Modal/ModalService.test.tsx
PASS src/views/Connection/ConnectionForm/ConnectionForm.test.tsx
PASS src/views/Connector/ServiceForm/ServiceForm.test.tsx (7.622 s)

Test Suites: 1 skipped, 20 passed, 20 of 21 total
Tests:       2 skipped, 121 passed, 123 total
Snapshots:   16 passed, 16 total
Time:        8.508 s, estimated 21 s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.4/userguide/command_line_interface.html#sec:command_line_warnings

Execution optimizations have been disabled for 2 invalid unit(s) of work during this build to ensure correctness.
Please consult deprecation warnings for more details.

BUILD SUCCESSFUL in 22s
373 actionable tasks: 5 executed, 368 up-to-date
```

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/airbytehq/airbyte?pullRequest=15074) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/airbytehq/airbyte?pullRequest=15074) before we can accept your contribution.<br/>**14** out of **15** committers have signed the CLA.<br/><br/>:white_check_mark: drewrasm<br/>:white_check_mark: marcosmarxm<br/>:white_check_mark: perangel<br/>:white_check_mark: grishick<br/>:white_check_mark: artem1205<br/>:white_check_mark: tuliren<br/>:white_check_mark: lazebnyi<br/>:white_check_mark: mfsiega-airbyte<br/>:white_check_mark: darynaishchenko<br/>:white_check_mark: subodh1810<br/>:white_check_mark: YowanR<br/>:white_check_mark: tybernstein<br/>:white_check_mark: Amruta-Ranade<br/>:white_check_mark: cuyk<br/>:x: github-actions[bot]<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/airbytehq/airbyte?pullRequest=15074) it.</sub>

marcosmarxm commented on airbyte-integrations/connectors/source-pinterest/source_pinterest/source.py:
```diff
@@ -251,8 +251,13 @@ def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
 
 
 class Campaigns(ServerSideFilterStream):
+    def __init__(self, parent: HttpStream, with_data_slices: bool = True, status_filter: str = "", **kwargs):
```
status_filter is an Array/List data type.

marcosmarxm commented on airbyte-integrations/connectors/source-pinterest/source_pinterest/spec.json:
```diff
@@ -36,6 +36,16 @@
+      },
+      "status": {
+        "title": "Status",
+        "description": "Statuses based off of your Pinterest entity. If you do not have a status set, it will be ignored completely.",
```
This is going to impact only a subset of streams? Let this clear here.

drewrasm replied:
Thanks for catching that, I adjusted the description in the latest commit.

marcosmarxm commented:
@drewrasm please sign the CLA.

drewrasm commented:
> @drewrasm please sign the CLA.

Just want to let you know I have signed the CLA, but I'm waiting for the status to change in the PR. Let me know if there's more steps I'm missing.

marcosmarxm commented:
@misteryeo can you check the change in the connector specification?

drewrasm commented:
@marcosmarxm let me know if there's anything in your requested changes I need to address, thanks! 👍

marcosmarxm commented:
/test connector=connectors/source-pinterest

drewrasm commented:
/test connector=connectors/source-pinterest

drewrasm commented:
@marcosmarxm I had the same failures in the tests happening independent if I was on my branch or master locally. I fixed what I could locally of the tests that were broken. Can you verify my changes broke the tests and that they weren't already failing? That would help out tremendously, thanks!

marcosmarxm commented:
I think our account is not passing the tests, I need to investigate further but asap I discovered that I'll return to you. Thanks for the contribution

drewrasm commented:
@marcosmarxm thanks for checking into that, is there any update towards getting those tests resolved? Please don't hesitate to let me know of anything I'll need to change in the PR. Thanks again!

drewrasm commented:
@misteryeo would you be able to give us an update on the progress of getting the tests resolved? Thank you for your time!

@wgormley
@coltonpomeroy

misteryeo commented:
Apologies for the delay here! Checking in with @marcosmarxm on this as he's the one investigating and will come back to you with an update!

marcosmarxm commented:
Sorry delay here @drewrasm I'll take a look tomorrow!

coltonpomeroy commented:
Hey @marcosmarxm, just wanted to check-in and see if you were able to take a look?

marcosmarxm commented:
Yes @coltonpomeroy but I'm having problem with the API token access. Hope get this solved today/tomorrow. Sorry the delay to fix it and merge your contribution

coltonpomeroy commented:
Hey @marcosmarxm, were you able to resolve the issue you were having with the API token access?

marcosmarxm committed:
solve conflicts

marcosmarxm committed:
solve conflict in json schema

marcosmarxm commented:
/test connector=connectors/source-pinterest

marcosmarxm commented:
> Hey @marcosmarxm, were you able to resolve the issue you were having with the API token access?

Yes. I'm trying to test the changes here.

marcosmarxm commented:
Sorry the delay here @drewrasm and @coltonpomeroy; the Pinterest source tests are not passing from previous code. I asked the team to correct that and after that I'll be able to test/publish your contribution.

coltonpomeroy commented:
Hey @marcosmarxm, any news from the team on these failed tests?

marcosmarxm commented:
@YowanR can you help here?

YowanR commented:
@marcosmarxm Sure! @igrankova and @ycherniaiev could you take a look at this? We are running into errors with API token access.

drewrasm commented:
Hello @YowanR, do you have any updates on the problems with the API token access?

coltonpomeroy commented:
@YowanR @marcosmarxm Just checking in to see if there's been any progress here?

YowanR commented:
@arsenlosenko Can you please take a look at this? We should prioritize this asap.
cc @lazebnyi

coltonpomeroy commented:
@arsenlosenko @lazebnyi Any updates on this?

arsenlosenko commented:
@YowanR @coltonpomeroy I am gonna take a look at the problem with tests, the test suite may fail because of the rate limits that Pinterest API has, when the tests are executed we might be above this limit

coltonpomeroy commented:
@arsenlosenko Is there a workaround in the event that we're blocked by rate limits with the Pinterest API?

drewrasm commented:
@arsenlosenko do you have an update on the tests?

arsenlosenko commented:
@drewrasm I wanted to to run tests for these specific changes in isolation locally, but I was busy with other tasks at the moment, going to do that either today or on Monday, I'll let you know how it goes. So far, I think that's the only way to test this without triggering the rate limits, but I will check what else could be done here.

coltonpomeroy commented:
@arsenlosenko Thanks for the update and your work on these tests. We're really looking forward to getting the PR merged.

coltonpomeroy commented:
@arsenlosenko Did we end up running into issues with getting rate limited?

arsenlosenko commented:
@coltonpomeroy hi, I was busy with other tasks at the moment, coming back to this PR once again, will let you know when the tests for changed streams were executed locally, and I'll provide the results of these tests

coltonpomeroy commented:
@arsenlosenko thanks for letting me know, do you have any timeline for when that should be done?

arsenlosenko commented:
@coltonpomeroy need to complete one more task with higher priority at the moment, it should be done today, and then I'll come back to this PR, I want to complete it until next afternoon

drewrasm committed:
Merge branch 'master' into ar/feat_pinterest_filters

drewrasm committed:
bump to version 0.1.8 for the changes of this pr

arsenlosenko commented:
> @arsenlosenko Just wanted to check-in on this. We're hoping to release a feature in our product soon that is dependent on this PR getting merged. Do you have an updated timeline for when that might be possible?

@coltonpomeroy today and on Friday I have worked on testing of this PR, but each time I exceeded the rate limit of Pinterest API before I managed to isolate tests to test specific changes to this PR. I came up with another approach that I will try tomorrow, I'll try to complete it ASAP

arsenlosenko commented:
/test connector=connectors/source-pinterest

arsenlosenko commented:
@coltonpomeroy I have approved the PR, tried to run tests for Pinterest again to see how they perform, but right now we are blocked due to broken tests in master branch which affects tests here. Once this issue is fixed, I am going to continue working on this PR.

arsenlosenko commented:
@drewrasm can you please update your fork with latest changes in the repo? yesterday there was a problem with test execution, it was fixed already, but in order to run tests properly in your PR the fork should also be updated. I can do the update myself from UI here in PR though, but it would be better to do it from your side to not cause any additional problems.

drewrasm committed:
Merge branch 'master' into ar/feat_pinterest_filters

arsenlosenko commented:
/test connector=connectors/source-pinterest

coltonpomeroy commented:
@arsenlosenko All merged, let us know if there's anything else you need from our end.

drewrasm committed:
Merge branch 'master' into ar/feat_pinterest_filters

drewrasm commented:
@arsenlosenko thank you for looking into this! I have it all up to date now, what's left for us to get this merged? Please let me know if there's anything left on my end, thanks!

drewrasm committed:
Merge branch 'master' into ar/feat_pinterest_filters

coltonpomeroy commented:
@arsenlosenko Anything else needed on our end in order to merge this PR?

arsenlosenko commented:
@coltonpomeroy @drewrasm thanks for the help, I will continue working on tests now, no additional help from you is needed, I'll keep you posted.

drewrasm committed:
Merge branch 'master' into ar/feat_pinterest_filters

drewrasm committed:
Merge branch 'master' into ar/feat_pinterest_filters

coltonpomeroy commented:
@arsenlosenko @marcosmarxm Looks like all workflow items are passing, are we able to get this merged?

drewrasm committed:
Merge branch 'master' into ar/feat_pinterest_filters

arsenlosenko commented:
/test connector=connectors/source-pinterest

drewrasm committed:
Merge branch 'master' into ar/feat_pinterest_filters

coltonpomeroy commented:
@arsenlosenko Just wanting to check-in here and get a status update. Please let us know if there's anything we can do on our end to help.

@marcosmarxm @misteryeo

arsenlosenko commented:
@coltonpomeroy hi, the tests have ran successfully, except backwards compatibility tests, working on the change that would make them pass, once it is all good, the PR can be finally merged.

drewrasm committed:
Merge branch 'master' into ar/feat_pinterest_filters

drewrasm committed:
Merge branch 'master' into ar/feat_pinterest_filters

coltonpomeroy commented:
Thanks for your work on that test @arsenlosenko, we're really looking forward to having the PR merged.

arsenlosenko commented on airbyte-integrations/connectors/source-pinterest/source_pinterest/schemas/ad_account_analytics.json:
```diff
@@ -7,7 +7,7 @@
     },
     "AD_ACCOUNT_ID": {
-      "type": ["null", "string"]
+      "type": ["null", "string", "number"]
```
@drewrasm can you please change the values here to just `["string"]`, as it stated in the [Pinterest documentation](https://developers.pinterest.com/docs/api/v5/#operation/ad_account/analytics) the value for `ad_account_id` should always be a string, and it is always required.

arsenlosenko commented on airbyte-integrations/connectors/source-pinterest/source_pinterest/schemas/ad_group_analytics.json:
```diff
@@ -7,7 +7,7 @@
     },
     "AD_ACCOUNT_ID": {
-      "type": ["integer"]
+      "type": ["null", "string", "number"]
```
same goes here, please change it to `["string"]`

arsenlosenko commented on airbyte-integrations/connectors/source-pinterest/source_pinterest/schemas/campaign_analytics.json:
```diff
@@ -7,7 +7,7 @@
     },
     "AD_ACCOUNT_ID": {
-      "type": ["null", "integer"]
+      "type": ["null", "string", "number"]
```
and here is the same, change it `["string"]` please, and then we will run the tests again, it should fix the problem we had with failed tests

drewrasm replied:
sweet, it's changed

drewrasm committed:
Merge branch 'master' into ar/feat_pinterest_filters

drewrasm committed:
change ad account id in the schemas

drewrasm committed:
Merge branch 'master' into ar/feat_pinterest_filters

drewrasm commented:
@arsenlosenko looks like that wasn't the issue, but I'm glad we fixed it. 
Even running tests on master locally, I can see that we get the same warnings/errors

SKIPPED [1] ../usr/local/lib/python3.9/site-packages/source_acceptance_test/plugin.py:63: Skipping TestFullRefresh.test_sequential_reads: not found in the config.
SKIPPED [1] ../usr/local/lib/python3.9/site-packages/source_acceptance_test/plugin.py:63: Skipping TestIncremental.test_two_sequential_reads: not found in the config.

drewrasm committed:
Merge branch 'master' into ar/feat_pinterest_filters

drewrasm commented:
If the errors are independent of the changes of this branch, is it possible for us to get this pushed?
@marcosmarxm @misteryeo @YowanR

drewrasm committed:
Merge branch 'master' into ar/feat_pinterest_filters

Commit:
query to include data plane attributes (#18531)

* query to include data plane attributes

* rename functions

* fix java build

* more renaming fix

tuliren committed:
Fix unit tests in source relational db (#18789)

* Fix unit tests

* Add extra test case for record count > 1

* Store record count in variable

Commit:
ci: use custom test-reporter action to upload job results (#18004)

* ci: use custom action to upload job results

marcosmarxm committed:
Correct coinmarket spec (#18790)

* correct coinmarket spec

* remove duplicate support normalization from source spec

* rollback coinmarketcap version in source def seed

* update connector version

* auto-bump connector version

Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

grishick committed:
Parameterize test_empty_streams and test_stream_with_1_airbyte_column by destination (#18197)

* Remove lines that always add Postgres to list of destinations
* Parameterize all tests in test_ephemeral by destination

artem1205 committed:
🐛 Source Facebook Marketing: reduce request limit after specific error (#18734)

* 🐛 Source Facebook Marketing: reduce request limit after specific error

* 🐛 Source Facebook Marketing: bump version; update docs

* 🐛 Source Facebook Marketing: add test

* 🐛 Source Facebook Marketing: increase timeout

perangel committed:
[charts/airbyte-cron] Cleanup env vars (#18787)

* [charts/airbyte-cron] Cleanup env vars

* Remove unused env var

Commit:
Use equalsIgnoreCase (#18810)

github-actions[bot] committed:
Bump helm chart version reference to 0.40.40 (#18815)

Co-authored-by: perangel <perangel@users.noreply.github.com>
Co-authored-by: Kyryl Skobylko <xpuska513@gmail.com>

lazebnyi committed:
🐛Destination Google Sheets: Fix empty headers list (#18729)

* Fix empty headers list

* Updated PR number

* Bumped version

* auto-bump connector version

Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

lazebnyi committed:
🐛Source Exchange Rates: Fix handling error during check connection (#18726)

* Fix handling error during check connection

* Updated PR number

* auto-bump connector version

Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

grishick committed:
Add normalization changelog and bump normalization version in platform (#18813)

Commit:
Remove ConfigPersistence usage from SecretsMigrator (#18747)

Commit:
remove config persistence from seeding logic (#18749)

Commit:
Remove the bulk actions from ConfigPersistence (#18800)

Commit:
hide ConfigPersistence inside ConfigRepository to discourage use (#18803)

Commit:
ci: add job and run id to test reports (#18832)

octavia-squidington-iii committed:
Bump Airbyte version from 0.40.17 to 0.40.18 (#18827)

Co-authored-by: grishick <grishick@users.noreply.github.com>

Commit:
🪟🔧 Remove styled components (round 1) (#18766)

* refactor EditorHeader (untested)

* refactor BaseClearView

* delete unused Subtitle

* refactor ConfirmationModal

* refactor Arrow

* refactor BulkHeader

* refactor CatalogTreeSearch

* refactor StreamFieldTable

* refactor StreamHeader

* refactor ConnectorIcon

* refactor TreeRowWrapper

* refactor DeleteBlock

* refactor EmptyResourceBlock

* revert unintended element change

arsenlosenko commented:
@drewrasm good call, considering that it should be okay to merge this I think. If everyone is okay with that as well we should be good to go. 
One more thing to add though is to update the LABEL value in the Dockerfile with your version of connector (I believe it is version 0.1.9)

darynaishchenko committed:
fixed acceptance tests (#18699)

Commit:
🪟🔧 Reactor Breadcrumbs component to use anchors (#18764)

* refactor breadcrumbs to use actual links

* PR comments on styles

mfsiega-airbyte committed:
increase test timeout for some webapp tests to prevent flakes (#18807)

tybernstein committed:
Remove "Filters and Segments" from Google Analytics v4 (#18508)

Filters and Segments info was incorrectly added to the Google Analytics v4 connector instead of the Google Analytics (Universal Analytics) Connector.

YowanR committed:
Add notes about EU OAUth (#18835)

EU OAuth is not fully tested so adding a note to account for that.

Commit:
🪟🐛 Fix: visual regression in ConnectorIcon (#18849)

* fix visual regression

* remove unused prop

Commit:
Add links to demo page (#18828)

subodh1810 committed:
mysql-source:fix tinyint unsigned handling (#18619)

* mysql-source:fix tinyint unsigned handling

* update doc

* format

* upgrade version

* auto-bump connector version

Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

Commit:
:window: :tada: Allow environment specific sections in docs (#18829)

* Allow environment specific sections in docs

* Change syntax to lower case

cuyk committed:
ci: replace GITHUB_OUTPUT with GITHUB_ENV on multiline variables (#18809)

* ci: replace GITHUB_OUTPUT with GITHUB_ENV on multiline variables

* ci: replace github set-ouput with new syntax in ./tools/bin/

Commit:
Add connection ID to span (#18852)

Amruta-Ranade committed:
edited connector docs (#18855)

Commit:
:window: :wrench: Upgrade husky to 8.0.1 (#18719)

* Upgrade Husky

* Upgrade Husky

* Upgrade Husky

* Upgrade Husky

* Upgrade Husky

cuyk committed:
ci: replace GITHUB_OUTPUT with GITHUB_ENV for multiline variables (#18857)

Commit:
Avoid NPE when adding connection ID to trace (#18856)

Commit:
Filter exit errors by operation name (#18850)

drewrasm committed:
add label

drewrasm committed:
Merge branch 'master' into ar/feat_pinterest_filters

drewrasm committed:
Merge branch 'master' into ar/feat_pinterest_filters

drewrasm commented:
@arsenlosenko 
I changed the dockerfile and pulled from most recent master. It seems to have automatically done some rebasing, which added all those new labels. 
Can we get this merged today with the change to the dockerfile?

arsenlosenko commented:
/publish connector=connectors/source-pinterest

arsenlosenko commented:
@drewrasm thanks for added changes and keeping this PR up to date compared to master branch, I am publishing this version of connector now, once it is all done, the PR will be merged.

drewrasm committed:
Merge branch 'master' into ar/feat_pinterest_filters

drewrasm commented:
@arsenlosenko thank you!
I see that the publish command was run, is there anything you need from us?

drewrasm committed:
Merge branch 'master' into ar/feat_pinterest_filters

drewrasm committed:
Merge branch 'master' into ar/feat_pinterest_filters

arsenlosenko commented:
/publish connector=connectors/source-pinterest run-tests=false

arsenlosenko commented:
@drewrasm the previous command failed because of the tests, skipping them now and publishing again

octavia-squidington-iii committed:
auto-bump connector version

arsenlosenko commented:
@drewrasm the changes were published, will merge it once the required checks have passed

arsenlosenko commented:
@marcosmarxm I see that you have requested changes in this PR earlier, the PR is ready to be merged, can you please take a look if your requested changes were done?

drewrasm committed:
Merge branch 'master' into ar/feat_pinterest_filters

drewrasm commented:
@arsenlosenko Awesome, that's great news! Those requested changes should be taken care of. I have comments about how I resolved those requests. 
The PR looks all ready to me 👍
